mistralrs-core/src/pipeline/gguf_tokenizer.rs

```diff
@@ -197,13 +304,35 @@ mod tests {
         }
     }
 
-    #[allow(dead_code)]
+    // Content based upon https://github.com/ggerganov/llama.cpp/blob/master/tests/test-tokenizer-random.py#L99-L161
     fn get_test_passage() -> String {
-        let passage = reqwest::blocking::get("https://loripsum.net/api")
-            .expect("Failed to download sample text")
-            .bytes()
-            .expect("Failed to get bytes");
-        String::from_utf8(passage.to_vec()).expect("Failed to convert sample text to string.")
+        let passage = "Hello, world! \n🚀 (normal) 😶‍🌫️ (compound emoji, zwj sequence) ✅ (emoji as single token)\n你好世界！\nNǐ hǎo shìjiè!";
+
+        passage.to_owned()
+    }
+
```

Comment:
I think the emoji one is a bit interesting for test coverage. Especially the compound/ZWJ type.

👨‍👩‍👧‍👦 for example is 25 bytes, trimming some from the start or end can either produce a different emoji or corrupt output.

Reply:
Nice!